includes/embeds/class-amp-tiktok-embed-handler.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Class AMP_Tiktok_Embed_Handler
+ *
+ * @package AMP
+ */
+
+use Amp\AmpWP\Dom\Document;
+
+/**
+ * Class AMP_Tiktok_Embed_Handler
+ */
+class AMP_Tiktok_Embed_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Registers embed.
+	 */
+	public function register_embed() {
+		// Not implemented.
+	}
+
+	/**
+	 * Unregisters embed.
+	 */
+	public function unregister_embed() {
+		// Not implemented.
+	}
+
+	/**
+	 * Sanitize Tiktok embeds to be AMP compatible.
+	 *
+	 * @param Document $dom DOM.
+	 */
+	public function sanitize_raw_embeds( Document $dom ) {
+		$nodes = $dom->xpath->query( '//blockquote[ @class="tiktok-embed" ]' );
+
+		foreach ( $nodes as $node ) {
+			if ( ! $this->is_raw_embed( $node ) ) {
+				continue;
+			}
+
+			$this->make_embed_amp_compatible( $node );
+		}
+	}
+
+	/**
+	 * Determine if the node has already been sanitized.
+	 *
+	 * @param DOMElement $node The DOMNode.
+	 * @return bool Whether the node is a raw embed.
+	 */
+	protected function is_raw_embed( DOMElement $node ) {
+		return ! $node->firstChild || ( $node->firstChild && 'amp-iframe' !== $node->firstChild->nodeName );
+	}
+
+	/**
+	 * Make Tiktok embed AMP compatible.
+	 *
+	 * @param DOMElement $node The DOMNode to make AMP compatible.
+	 */
+	protected function make_embed_amp_compatible( DOMElement $node ) {
+		$dom       = $node->ownerDocument;
+		$embed_url = $node->getAttribute( 'cite' );
+		$video_id  = $node->getAttribute( 'data-video-id' );
+
+		$this->remove_embed_script( $node );
+
+		// Create fallback node.
+		$fallback_node = AMP_DOM_Utils::create_node(
+			$dom,
+			'a',
+			[
+				'href'  => esc_url_raw( $embed_url ),
+				'class' => 'amp-wp-embed-fallback',
+			]
+		);
+		$fallback_node->appendChild( new DOMText( esc_html( $embed_url ) ) );
+
+		// If there is no video ID, replace the blockquote with the fallback and return.
+		if ( empty( $video_id ) ) {
+			$node->parentNode->replaceChild( $fallback_node, $node );
+			return;
+		}
+
+		// Find existing <section> node to use as the placeholder.
+		$placeholder_node = null;
+		foreach ( iterator_to_array( $node->childNodes ) as $child ) {
+			if ( ! ( $child instanceof DOMElement ) ) {
+				continue;
+			}
+
+			if ( 'section' === $child->nodeName ) {
+				$placeholder_node = $node->removeChild( $child );
+				break;
+			}
+		}
+
+		// If the placeholder was not found, use the fallback instead.
+		if ( ! $placeholder_node ) {
+			$placeholder_node = $fallback_node;
+		}
+
+		$placeholder_node->setAttribute( 'placeholder', '' );
+
+		$amp_iframe_node = AMP_DOM_Utils::create_node(
+			$dom,
+			'amp-iframe',
+			[
+				'layout'  => 'responsive',
+				'width'   => $this->DEFAULT_WIDTH,
+				'height'  => $this->DEFAULT_HEIGHT,
```

Comment:
Should these note be derived from the `style="max-width: 605px;min-width: 325px;"` on the `blockquote`? Those properties should then be removed from the `blockquote`. 

Otherwise, the `blockquote` should get a `position:relative` and the `amp-iframe` should perhaps get `layout=fill`, no?

Reply:
The `blockquote` and `iframe` are resized during page load, so the width and height can't be derived from the `blockquote` style.

> Otherwise, the `blockquote` should get a `position:relative` and the `amp-iframe` should perhaps get `layout=fill`, no?

With that approach the `blockquote` nor `amp-iframe` is resized, leading to an almost non-visible embed:

![image](https://user-images.githubusercontent.com/16200219/75939184-66fefb00-5e81-11ea-8060-0670956e2894.png)